packages/webapp/src/containers/SensorReadings/saga.js

```diff
@@ -43,157 +43,193 @@ import i18n from '../../locales/i18n';
-          }
-          if (weatherURL === DAILY_FORECAST_API_URL) {
-            openWeatherUrl.searchParams.append('cnt', 1);
-          }
-          openWeatherPromiseList.push(call(axios.get, openWeatherUrl?.toString()));
-        }
+    for (let locationId of locationIds) {
+      sensorDataByLocationIds[locationId] = {};
+      for (let type of readingTypes) {
+        sensorDataByLocationIds[locationId][type] = {};
 
-        const [openWeatherResponse, predictedWeatherResponse, predictedDailyWeatherResponse] =
-          yield all(openWeatherPromiseList);
-        stationName = predictedDailyWeatherResponse?.data?.city.name;
-        const weatherResData = [
-          ...openWeatherResponse.data.list,
-          ...predictedWeatherResponse.data.list,
-        ];
+        let readings = sensorDataByLocationIds[locationId][type];
+        readings.selectedSensorName = data?.sensorsPoints[0]?.name;
+        readings.lastUpdatedReadingsTime = getLastUpdatedTime(
+          data?.sensorReading[type]
+            .filter((cv) => (cv.value ? cv.value : cv.value === 0))
+            .map((cv) => new Date(cv.actual_read_time).valueOf() / 1000),
+        );
+        readings.predictedXAxisLabel = roundDownToNearestChosenPoint(currentUnixTime);
 
-        const ambientData = weatherResData.reduce((acc, tempInfo) => {
-          let dateAndTimeInfo = new Date(tempInfo?.dt * 1000).toString();
-          const isCorrectTimestamp = GRAPH_TIMESTAMPS?.find((g) => dateAndTimeInfo?.includes(g));
-          if (isCorrectTimestamp && startDate < tempInfo?.dt && tempInfo?.dt < predictedEndDate) {
-            const currentDateTime = `${dateAndTimeInfo?.split(':00:00')[0]}:00`;
-            if (!isFound && currentDT < tempInfo?.dt) {
-              isFound = true;
-              predictedXAxisLabel = acc[Object.keys(acc).at(-1)][CURRENT_DATE_TIME];
-            }
-            if (!acc[tempInfo?.dt]) acc[tempInfo?.dt] = {};
-            acc[tempInfo?.dt] = {
-              ...acc[tempInfo?.dt],
-              [`${ambientTempFor} ${stationName}`]: tempInfo?.main?.temp,
+        // reduce sensor data
+        let typeReadings = {};
+        typeReadings = data?.sensorReading[type].reduce((acc, cv) => {
+          const currentValueUnixDate = new Date(cv?.read_time).getTime() / 1000;
+          const currentValueUnixTime = new Date(currentValueUnixDate * 1000).toString();
+          const isChosenTimestamp = CHOSEN_GRAPH_DATAPOINTS?.find((g) =>
+            currentValueUnixTime?.includes(g),
+          );
+          if (
+            isChosenTimestamp &&
+            historicalUnixDate < currentValueUnixDate &&
+            currentValueUnixDate < predictedUnixDate
+          ) {
+            const currentDateTime = `${currentValueUnixTime?.split(':00:00')[0]}:00`;
+            if (!acc[currentValueUnixDate]) acc[currentValueUnixDate] = {};
+            acc[currentValueUnixDate] = {
+              ...acc[currentValueUnixDate],
```

Comment:
Yes I think you are right since the reduce function does not take in a starter accumulator like the weather data does and needs to.

Removed, good find!